src/test/scala/edu/ie3/simona/model/participant/load/FixedLoadModelSpec.scala

```diff
@@ -0,0 +1,151 @@
+    with DefaultTestData
+    with TableDrivenPropertyChecks {
+
+  private implicit val tolerance: Power = Watts(1d)
+
+  "A fixed load model" should {
+
+    val defaultOperationInterval =
+      SystemComponent.determineOperationInterval(
+        defaultSimulationStart,
+        defaultSimulationEnd,
+        loadInput.getOperationTime,
+      )
+
+    "be instantiated from valid input correctly" in {
+      val testData = Table(
+        ("reference", "expectedReferenceActivePower"),
+        (ActivePower(Watts(268.6)), Watts(268.6)),
+        (EnergyConsumption(KilowattHours(3000d)), Watts(342.24)),
+      )
+
+      forAll(testData) { (reference, expectedReferenceActivePower: Power) =>
+        val actual = new FixedLoadModel(
+          loadInput.getUuid,
+          loadInput.getId,
+          defaultOperationInterval,
+          QControl.apply(loadInput.getqCharacteristics),
+          Kilowatts(
+            loadInput.getsRated
+              .to(PowerSystemUnits.KILOWATT)
+              .getValue
+              .doubleValue()
+          ),
+          loadInput.getCosPhiRated,
+          reference,
+        )
+
+        val calculatedPower = actual
+          .calculateActivePower(
+            ModelState.ConstantState,
+            FixedLoadModel.FixedLoadRelevantData,
+          )
+
+        calculatedPower should approximate(expectedReferenceActivePower)
+      }
+    }
+
+    "return approximately the same power in 10,000 calculations" in {
+
+      val testData = Table(
+        ("reference", "expectedPower"),
+        (ActivePower(Watts(268.6)), Watts(268.6)),
+        (EnergyConsumption(KilowattHours(3000d)), Watts(342.24)),
+      )
+
+      forAll(testData) { (reference, expectedPower: Power) =>
+        val dut = new FixedLoadModel(
+          loadInput.getUuid,
+          loadInput.getId,
+          defaultOperationInterval,
+          QControl.apply(loadInput.getqCharacteristics),
+          Kilowatts(
+            loadInput.getsRated
+              .to(PowerSystemUnits.KILOWATT)
+              .getValue
+              .doubleValue()
+          ),
+          loadInput.getCosPhiRated,
+          reference,
+        )
+
+        for (_ <- 0 until 10000) {
+          val calculatedPower = dut
+            .calculateActivePower(
+              ModelState.ConstantState,
+              FixedLoadModel.FixedLoadRelevantData,
+            )
+
+          calculatedPower should approximate(expectedPower)
+        }
+      }
+    }
+
+    "consider the (global) scaling factor correctly" in {
+      val testData = Table(
+        ("reference", "expectedPower"),
+        (ActivePower(Watts(268.6d)), Watts(268.6)),
+        (EnergyConsumption(KilowattHours(3000d)), Watts(342.24)),
+      )
+
+      forAll(testData) { (reference, expectedPower: Power) =>
+        val relevantData = FixedLoadModel.FixedLoadRelevantData
+
+        var scale = 0.0
+        while (scale <= 2) {
+          val scaledSRated = Kilowatts(
```

Comment:
As Sebastian stated correctly: Using var is usually not what we want, but rather immutable values. We should look into this and try to find such a solution. Maybe this can be done within #886 or as an extra issue / PR

Reply:
Opened up issue here: #919. Could be solved together with the issue that Daniel mentioned or separately